connection.go

```diff
@@ -696,14 +696,16 @@ func (c *Connection) readFrames(_ uint32) {
 func (c *Connection) handleFrameRelay(frame *Frame) bool {
 	switch frame.Header.messageType {
 	case messageTypeCallReq, messageTypeCallReqContinue, messageTypeCallRes, messageTypeCallResContinue, messageTypeError:
-		if err := c.relay.Relay(frame); err != nil {
+		var shouldRelease bool
+		var err error
+		if shouldRelease, err = c.relay.Relay(frame); err != nil {
 			c.log.WithFields(
 				ErrField(err),
 				LogField{"header", frame.Header},
 				LogField{"remotePeer", c.remotePeerInfo},
 			).Error("Failed to relay frame.")
 		}
-		return false
+		return shouldRelease
```

Comment:
the nested if is useful for scoping variables to the if, but in this case, we're not scoping any variables, so it'll be cleaner to avoid it,
```
		shouldRelease, err := c.relay.Relay(frame)
		if err != nil {
			c.log.WithFields(				c.log.WithFields(
				ErrField(err),					ErrField(err),
				LogField{"header", frame.Header},					LogField{"header", frame.Header},
				LogField{"remotePeer", c.remotePeerInfo},					LogField{"remotePeer", c.remotePeerInfo},
			).Error("Failed to relay frame.")				).Error("Failed to relay frame.")
		}			}
		return false			return shouldRelease

```